apps/meteor/tests/e2e/messaging.spec.ts

```diff
@@ -22,121 +22,184 @@ test.describe.serial('Messaging', () => {
+			await page.keyboard.press('Tab');
+			await page.keyboard.press('Space');
+			await expect(poHomeChannel.userCardToolbar).toBeVisible();
+
+			// close UserCard with Esc
+			await page.keyboard.press('Escape');
+			await expect(poHomeChannel.userCardToolbar).not.toBeVisible();
+
+			// with focus restored reopen toolbar
+			await page.keyboard.press('Space');
+			await expect(poHomeChannel.userCardToolbar).toBeVisible();
+
+			// close UserCard with button
+			await page.keyboard.press('Tab');
+			await page.keyboard.press('Tab');
+			await page.keyboard.press('Tab');
+			await page.keyboard.press('Space');
+			await expect(poHomeChannel.userCardToolbar).not.toBeVisible();
+		});
 
-		await expect(page.locator('[data-qa-type="message"]:has-text("msg2")')).toBeFocused();
-	});
+		test('should not restore focus on the last focused if it was triggered by click', async ({ page }) => {
+			await poHomeChannel.sidenav.openChat(targetChannel);
+			await page.locator('[data-qa-type="message"]:has-text("msg1")').click();
+			await poHomeChannel.composer.click();
+			await page.keyboard.press('Shift+Tab');
+
+			await expect(page.locator('[data-qa-type="message"]:has-text("msg2")')).toBeFocused();
+		});
+
+		test('should not focus on the last message when focusing by click', async ({ page }) => {
+			await poHomeChannel.sidenav.openChat(targetChannel);
+			await page.locator('[data-qa-type="message"]:has-text("msg1")').click();
+
+			await expect(page.locator('[data-qa-type="message"]').last()).not.toBeFocused();
+		});
+
+		test('should focus the latest message when moving the focus on the list and theres no previous focus', async ({ page }) => {
+			await poHomeChannel.sidenav.openChat(targetChannel);
+			await page.getByRole('button', { name: targetChannel }).first().focus();
+
+			// move focus to the list
+			await page.keyboard.press('Tab');
+			await page.keyboard.press('Tab');
+			await page.keyboard.press('Tab');
+			await expect(page.locator('[data-qa-type="message"]').last()).toBeFocused();
 
-	test('should not focus on the last message when focusing by click', async ({ page }) => {
-		await poHomeChannel.sidenav.openChat(targetChannel);
-		await page.locator('[data-qa-type="message"]:has-text("msg1")').click();
+			await page.getByRole('button', { name: targetChannel }).first().focus();
 
-		await expect(page.locator('[data-qa-type="message"]').last()).not.toBeFocused();
+			// move focus to the list again
+			await page.keyboard.press('Tab');
+			await page.keyboard.press('Tab');
+			await page.keyboard.press('Tab');
+			await expect(page.locator('[data-qa-type="message"]').last()).toBeFocused();
+		});
 	});
 
-	test('should focus the latest message when moving the focus on the list and theres no previous focus', async ({ page }) => {
-		await poHomeChannel.sidenav.openChat(targetChannel);
-		await page.getByRole('button', { name: targetChannel }).first().focus();
+	test.describe.serial('Message by "chat.postMessage" API method', () => {
```

Comment:
I think this test could be flaky, since between the post message and the expect, it can take some time, the expect could in some cases get the previous message, as it is using the last message locator. Maybe await for a confirmation that the message you want to test is already on the screen?